recipes/xz_utils/all/conanfile.py

```diff
@@ -25,7 +25,7 @@ def _is_mingw_windows(self):
         return self.settings.compiler == 'gcc' and self.settings.os == 'Windows' and os.name == 'nt'
 
     def build_requirements(self):
-        if self._is_mingw_windows:
+        if self._is_mingw_windows and "CONAN_BASH_PATH" not in os.environ:
```

Comment:
we also sometimes use more complicated condition:
 `"CONAN_BASH_PATH" not in os.environ and tools.os_info.detect_windows_subsystem() `
do you think if it's more reliable? in which cases it is needed?

Reply:
@uilianries ping

Reply:
@SSE4 is right, this case is more specific. Checking msys2 can avoid possible problems related to misconfiguration, like it's msys2 but is not detect as is.

Reply:
Done! I've added @SSE suggestion. Thanks for reviewing!